MovingBlocks/Terasology
# chore(ModuleManager): refactor and document engine module setup

This is the code that has been sitting around in my working directory that was ostensibly to fix #4639, the failures detected by MaliciousModule.

Upon closer inspection, it seems that https://github.com/MovingBlocks/gestalt/pull/112 is sufficient to fix that error on its own. These are changes (mostly refactors) I did while trying to determine how to write tests or fix the problem without reverting the code in gestalt. They may not be functionally _necessary,_ but maybe worth keeping for de-tangling this code.

This doesn't influence the MTE test failures either. That remains an independent problem.

### How to test

- normal game-with-modules stuff
- start game with MaliciousModule and look in log for messages from MaliciousTestSystem. Ensure it says 2/2 tests passed.

### Outstanding before merging

- [x] merge https://github.com/MovingBlocks/gestalt/pull/112
